### ISSUE TITLE
Base64 text encoding issue (Python 3)

### ISSUE THREAD

**Issue by katger4** [search5/solrpy#51]
Using solrpy 0.9.9, Python 3.7.1:

There is an issue with the text encoding for http authentication in this version. 

When creating a solr object, e.g. `solr.Solr(baseURL, http_user=httpuser, http_pass=httppass)`, core.py produces the following error at line 427 `http_auth = 'Basic ' + http_auth.encode('base64').strip()`:
`LookupError: 'base64' is not a text encoding; use codecs.encode() to handle arbitrary codecs`

http_user and http_pass are str type 
I don't get this error using Python 2.7

I got it to work by importing the base64 module and changing line 427 to: 
`http_auth = b'Basic ' + base64.b64encode(http_auth.encode()).strip()`

**Comment by search5** [comment]
The answer is late. I will apply the 1.0.0 version to your problem.

Thank you.

**Comment by katger4** [comment]
No worries!

I tried to use the latest version (1.0.0) with Python 3.7.1 and I now get the following error when creating a solr object, e.g. `solr.Solr(baseURL, http_user=httpuser, http_pass=httppass)`:

at line 432 in core.py: `http_auth = 'Basic ' + base64.b64encode(http_auth)`

`TypeError: can only concatenate str (not "bytes") to str` 

because `'Basic'` is `str` and `http_auth` is `bytes`

**Comment by nrshapiro** [comment]
Yes, looking at the proposed solution, there is a problem as others have said above.  Question though: how can we use the pre-release versions for testing, and where are they (I didn't see them on github, but maybe I'm not looking in the right place.)  Hopefully, this will be fixed and loaded to Pypi soon.  Thanks for your efforts in keeping Solrpy going!  It's important to us.
